nucypher/blockchain/eth/interfaces.py

```diff
@@ -253,6 +253,8 @@ def __init__(
         self.gas_strategy = gas_strategy or self.DEFAULT_GAS_STRATEGY
         self.max_gas_price = max_gas_price
 
+        self.__is_connected = False
```

Comment:
```suggestion
        self._is_connected = False
```
Double leading underscore has a specific meaning in python and triggers [Name Mangling](https://realpython.com/python-double-underscore/#understanding-name-mangling). For internal variables, stick with a single leading underscore.

Reply:
I think name mangling was intentional here, since we sometimes use this convention throughout the codebase to signal private attribute/methods (just search for `self.__` in your IDE), but I'm wondering if we may be misusing it. I'm not a super-seasoned Pythonista so I'll let others chime in.

Reply:
Given the sensitivity of the functionality gated by this variable the name mangling was deliberate here as we *really* do not want unintentional mutation of this variable (programmatically or explicitly).  In essence this is indeed a "private" meant to be used only by the `connect` method.

Reply:
☝️

Reply:
but this attribute doesn't exist on the parent class, so there isn't any danger of a name clash

Reply:
It's not really about a name clash - it's more about preventing any unintended modification to this variable by external entities (sub-classes or otherwise). It's a safety measure that if a dev is trying to change the value of that variable directly, you have to explicitly use the name mangling, and not just the plain variable name, which should give the dev a signal that they probably should not be doing that.

Reply:
Fair enough, but this is pretty un-pythonic :) In python, the signal that they probably shouldn't do that, is a leading underscore. It's just a cultural thing between Python and Java.